### PR TITLE
Add simplified testing instructions, testing detail, etcd check

### LINKED ISSUE #14370: Document how to write good tests
* Unit tests (table driven, etc.)
* Integration tests (including dealing with issues like multiple tests running concurrently)
* e2e tests

We have a few testing conventions here:
https://github.com/kubernetes/kubernetes/blob/master/docs/devel/coding-conventions.md

We want more tests, but not flaky tests.

cc @quinton-hoole @saad-ali @thockin @lavalamp

### PR DESCRIPTION
**Pull request by fejta**
Closes #14370 

In development.md add basic testing instructions.

Add a testing.md with details about unit, integration and e2e testing, including common conventions.

Add a check to hack/test-integration.sh to make sure that etcd is in PATH before starting, instead of failing later down the line.

See https://github.com/fejta/kubernetes/blob/testing/docs/devel/testing.md

### CODE REVIEW

File: docs/devel/development.md [hunk 1]
```diff
@@ -230,83 +230,18 @@ Please send dependency updates in separate commits within your PR, for easier re
 
 6) If you updated the Godeps, please also update `Godeps/LICENSES` by running `hack/update-godep-licenses.sh`.
 
+## Testing
 
-## Unit tests
+Three basic commands let you run unit, integration and/or unit tests:
```
Comment (thockin) [inline docs/devel/development.md]: you said "unit" twice

File: docs/devel/testing.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Testing guide
+
+This assumes you already read the [development guide](development.md) to
+install go, godeps and configure your git client.
+
+In order to send pull requests you need to make sure you changes pass
+unit, integration tests.
```
Comment (thockin) [inline docs/devel/testing.md]: "unit and integration"

File: docs/devel/testing.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Testing guide
+
+This assumes you already read the [development guide](development.md) to
+install go, godeps and configure your git client.
```
Comment (thockin) [inline docs/devel/testing.md]: Oxford comma after godeps?

File: docs/devel/testing.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+* Use ginko to desribe steps.
+  - See [should run a job to completion when tasks succeed](../../test/e2e/job.go)
+* Use [NewDefaultFramework](../../test/e2e/framework.go)
+  - Contains clients, namespace and auto resource cleanup
+* See [coding conventions](coding-conventions.md).
+
+### e2e test philosophy
+
+In general passing unit and integration tests should provide sufficient
+confidence
```
Comment (thockin) [inline docs/devel/testing.md]: weird line break

**Comment by thockin** [comment]
Just nits then LGTM.  I sort of hate that the command to run tests is different for different scenarios.  I will make it my personal mission to fix that by giving the one true interface to running stuff - make.

**Comment by thockin** [comment]
You can self LGTM after nits

File: docs/devel/development.md (hunk 1 above)
Comment (fejta) [inline docs/devel/development.md]: done

File: docs/devel/testing.md (hunk 2 above)
Comment (fejta) [inline docs/devel/testing.md]: done

File: docs/devel/testing.md (hunk 3 above)
Comment (fejta) [inline docs/devel/testing.md]: done

File: docs/devel/testing.md (hunk 4 above)
Comment (fejta) [inline docs/devel/testing.md]: done

**Comment by fejta** [comment]
thanks Tim!

**Pull request merged**